src/posts/introduction-webc.md

```diff
@@ -0,0 +1,236 @@
+
+```html
+<footer>
+  <p>&copy; 2022 Yours Truly</p>
+</footer>
+```
+
+Now we can use a `<site-footer></site-footer>` custom element and WebC will replace that element with the markup from `site-footer.webc`.
+
+> For the purposes of this article, we won’t concern ourselves with [importing components in WebC](https://github.com/11ty/webc#html-imports-kidding-kinda). When you set up the WebC plugin for Eleventy, you can [configure a directory for global components](https://www.11ty.dev/docs/languages/webc/#global-no-import-components) so that they’re always available.
+
+So if we have a file — `page.webc` — with all of our HTML boilerplate, we insert our footer into that file using this custom element.
+
+```html
+<!doctype html>
+<html>
+  <head></head>
+  <body>
+    <site-footer></site-footer>
+  </body>
+</html>
+```
+
+When we process `page.webc`, the resulting HTML will look like this (give or take some whitespace):
+
+```html
+<!doctype html>
+<html>
+  <head></head>
+  <body>
+    <footer>
+      <p>&copy; 2022 Yours Truly</p>
+    </footer>
+  </body>
+</html>
+```
+
+## Single file components
+
+At this point, it may still be unclear what advantages we get by using WebC. After all, many of the template languages that Eleventy already supports allow you to import partials from other files like this. But what WebC can add to even this basic use case is the single file component authoring experience.
+
+Single file components (SFC) are common in JavaScript frameworks such as Vue and Svelte. An SFC is a single file that contains HTML, CSS, and JavaScript. They often follow this structure:
+
+```html
+<script></script>
+
+<!-- Markup here -->
+
+<style></style>
+```
+
+Going back to our `site-footer.webc`, we can colocate the styles for our footer with the markup.
+
+```html
+<footer>
+  <p>&copy; 2022 Yours Truly</p>
+</footer>
+
+<style>
+  footer {
+    padding-block: 3rem;
+    padding-inline: 1rem;
+  }
+
+  footer > p {
+    font-family: fantasy;
+  }
+</style>
+```
+
+One advantage to writing your components this way instead of having your markup and styles in separate files is that when you’re markup changes, it’s easier to remember and update the styles. If we introduce a `<div>` to our footer to wrap the paragraph at some maximum width, our (admittedly contrived) `font-family` declaration will no longer apply to the copyright notice in the footer. Being able to see the markup and styles right next to each other helps us keep them synchronized.
+
+> When you have a component that is just HTML, WebC will replace the custom element with that HTML. But as soon as you introduce either a `<script>` tag or a `<style>` tag to the component, WebC will, by default, leave the custom element in place in case you are using it in either your scripts or your styles. You can tell WebC to remove the custom element by adding [webc:nokeep](https://www.11ty.dev/docs/languages/webc/#webcnokeep) to the custom element.
```

Comment:
Maybe switch this out of blockquote and add a headline like `### Custom element tag rendering` or something like that?